homeassistant/helpers/reload.py

```diff
@@ -120,7 +121,7 @@ async def _async_setup_platform(
         )
         return
 
-    entity_component: EntityComponent = hass.data[integration_platform]
+    entity_component: EntityComponent[Entity] = hass.data[integration_platform]
```

Comment:
I've used the base Entity class here because I wasn't quite sure.
Maybe I should be using # type: ignore[var-annotated] instead?

Reply:
base class is good